app/controllers/verify/finance_controller.rb

```diff
@@ -1,6 +1,7 @@
 module Verify
   class FinanceController < StepController
     before_action :confirm_step_needed
+    before_action :confirm_step_allowed
```

Comment:
they are discrete logical operations (requirement vs permission) so I wanted to keep them as separate methods.